File: adapters/go/schemas.go
package lh_adapter

// DNA_SCHEMA is the JSON schema for DNA traceability format validation.
var DNA_SCHEMA = map[string]interface{}{
	"$schema": "https://json-schema.org/draft/2020-12/schema",
	"title":   "LongHun DNA Traceability Format v∞",
	"type":    "string",
	"pattern": DNA_REGEX,
	"examples": []string{
		"#LongHun⚡️BingWu·GuiWei·JiaZi·WeiShi·䷾JiJi-ADAPTER-CODE-WRAP-V1.0-a3f8c1d9",
	},
}

// AUDIT_SCHEMA is the JSON schema for seven-factor behavioral audit records.
var AUDIT_SCHEMA = map[string]interface{}{
	"$schema": "https://json-schema.org/draft/2020-12/schema",
	"title":   "LongHun Seven-Factor Behavioral Audit v1.0",
	"type":    "object",
	"required": []string{"dna", "audit", "payload", "meta"},
	"properties": map[string]interface{}{
		"dna": map[string]interface{}{
			"type":        "string",
			"description": "DNA traceability identifier",
			"pattern":     DNA_REGEX,
		},
		"audit": map[string]interface{}{
			"type":     "object",
			"required": []string{"audit_version", "uid", "behavior_signature", "behavior_pattern", "behavior_labels", "color"},
			"properties": map[string]interface{}{
				"audit_version": map[string]interface{}{
					"type":    "string",
					"pattern": "^v\\d+\\.\\d+$",
				},
				"uid": map[string]interface{}{
					"type":    "string",
					"pattern": "^UID\\d+$",
				},
				"behavior_signature": map[string]interface{}{
					"type": "object",
					"required": []string{"P", "F", "T", "E", "C", "R", "A", "X", "Y", "Z"},
					"properties": map[string]interface{}{
						"P": map[string]interface{}{
							"type": "string",
							"enum": []string{"HasPromise", "NoPromise"},
						},
						"F": map[string]interface{}{
							"type": "string",
							"enum": []string{"Fulfilled", "Unfulfilled", "Partial"},
						},
						"T": map[string]interface{}{
							"type": "number",
						},
						"E": map[string]interface{}{
							"type": "string",
							"enum": []string{"Willing", "Perfunctory", "Resentful", "Numb"},
						},
						"C": map[string]interface{}{
							"type": "integer",
						},
						"R": map[string]interface{}{
							"type":    "integer",
							"minimum": 0,
						},
						"A": map[string]interface{}{
							"type": "string",
							"enum": []string{"Self", "Partner", "Family", "Outsider", "Public"},
						},
						"X": map[string]interface{}{
							"type": "string",
							"enum": []string{"OverExplain", "Silent", "Genuine", "Indifferent"},
						},
						"Y": map[string]interface{}{
							"type": "string",
							"enum": []string{"Changed", "Resisted", "Indifferent", "NoResponse"},
						},
						"Z": map[string]interface{}{
							"type": "number",
						},
					},
				},
				"behavior_pattern": map[string]interface{}{
					"type": "string",
					"enum": []string{
						"MODE-DefensiveDefaulter",
						"MODE-ExternalTrustSpender",
						"MODE-InternalDestroyer",
						"MODE-Fluctuating",
						"MODE-StableDisciplined",
					},
				},
				"behavior_labels": map[string]interface{}{
					"type":  "array",
					"items": map[string]interface{}{"type": "string"},
				},
				"color": map[string]interface{}{
					"type": "string",
					"enum": []string{"🟢", "🟡", "🔴"},
				},
			},
		},
		"payload": map[string]interface{}{
			"description": "Arbitrary payload data",
		},
		"meta": map[string]interface{}{
			"type":     "object",
			"required": []string{"adapter_version", "uid", "format"},
			"properties": map[string]interface{}{
				"adapter_version": map[string]interface{}{
					"type": "string",
				},
				"uid": map[string]interface{}{
					"type": "string",
				},
				"device": map[string]interface{}{
					"type": "string",
				},
				"task_type": map[string]interface{}{
					"type": "string",
				},
				"persona": map[string]interface{}{
					"type": "string",
				},
				"generated_at": map[string]interface{}{
					"type":   "string",
					"format": "date-time",
				},
				"format": map[string]interface{}{
					"type": "string",
				},
			},
		},
	},
}

File: adapters/ruby/lib/lh/schemas.rb
module Lh
  DNA_SCHEMA = {
    '$schema' => 'https://json-schema.org/draft/2020-12/schema',
    '$id' => 'https://uid9622.cn/schemas/dna-v1.0.json',
    'title' => 'LongHun DNA Traceability Code',
    'type' => 'object',
    'required' => %w[dna format uid timestamp],
    'properties' => {
      'dna' => { 'type' => 'string' },
      'format' => { 'type' => 'string', 'enum' => %w[v1.0 v2.0 v∞ compact] },
      'uid' => { 'type' => 'string', 'pattern' => '^UID\\d+$' },
      'device' => { 'type' => 'string' },
      'timestamp' => { 'type' => 'string', 'format' => 'date-time' },
    },
  }.freeze

  AUDIT_SCHEMA = {
    '$schema' => 'https://json-schema.org/draft/2020-12/schema',
    '$id' => 'https://uid9622.cn/schemas/audit-v1.0.json',
    'title' => 'LongHun Audit Record',
    'type' => 'object',
    'required' => %w[dna audit payload meta],
    'properties' => {
      'dna' => { 'type' => 'string' },
      'audit' => {
        'type' => 'object',
        'required' => %w[audit_version uid behavior_signature behavior_pattern behavior_labels color],
        'properties' => {
          'audit_version' => { 'type' => 'string' },
          'uid' => { 'type' => 'string' },
          'persona' => { 'type' => 'string' },
          'task_type' => { 'type' => 'string' },
          'behavior_signature' => {
            'type' => 'object',
            'required' => %w[P F T E C R A X Y Z],
            'properties' => {
              'P' => { 'enum' => %w[HasPromise NoPromise] },
              'F' => { 'enum' => %w[Fulfilled Unfulfilled Partial] },
              'T' => { 'type' => 'number' }, 'C' => { 'type' => 'number' },
              'E' => { 'enum' => %w[Willing Perfunctory Resentful Numb] },
              'R' => { 'type' => 'integer', 'minimum' => 0 },
              'A' => { 'enum' => %w[Self Partner Family Outsider Public] },
              'X' => { 'enum' => %w[OverExplain Silent Genuine Indifferent] },
              'Y' => { 'enum' => %w[Changed Resisted Indifferent NoResponse] },
              'Z' => { 'type' => 'number' },
            },
          },
          'behavior_pattern' => { 'enum' => Validator::VALID_PATTERNS },
          'behavior_labels' => { 'type' => 'array', 'items' => { 'type' => 'string' } },
          'color' => { 'enum' => %w[🟢 🟡 🔴] },
          'timestamp' => { 'type' => 'string', 'format' => 'date-time' },
          'payload_hash' => { 'type' => 'string', 'pattern' => '^[a-f0-9]{16}$' },
        },
      },
      'payload' => {},
      'meta' => {
        'type' => 'object',
        'required' => %w[adapter_version uid device task_type persona],
        'properties' => {
          'adapter_version' => { 'type' => 'string' }, 'uid' => { 'type' => 'string' },
          'device' => { 'type' => 'string' }, 'task_type' => { 'type' => 'string' },
          'persona' => { 'type' => 'string' }, 'generated_at' => { 'type' => 'string', 'format' => 'date-time' },
          'format' => { 'const' => 'longhun-v∞' },
        },
      },
    },
  }.freeze
end
